code/game/gamemodes/blob/powers.dm

```diff
@@ -31,246 +27,131 @@
 		if(chosen_node)
 			src.loc = chosen_node.loc
 
-/mob/camera/blob/verb/create_shield_power()
-	set category = "Blob"
-	set name = "Create Shield Blob (10)"
-	set desc = "Create a shield blob."
-
-	var/turf/T = get_turf(src)
-	create_shield(T)
-
-/mob/camera/blob/proc/create_shield(turf/T)
-
+/mob/camera/blob/proc/createSpecial(price, blobType, nearEquals, turf/T)
+	if(!T)
+		T = get_turf(src)
 	var/obj/effect/blob/B = (locate(/obj/effect/blob) in T)
-
-	if(!B)//We are on a blob
-		src << "There is no blob here!"
+	if(!B)
+		src << "<span class='warning'>There is no blob here!</span>"
 		return
-
 	if(!istype(B, /obj/effect/blob/normal))
-		src << "Unable to use this blob, find a normal one."
+		src << "<span class='warning'>Unable to use this blob, find a normal one.</span>"
 		return
-
-	if(!can_buy(10))
+	if(nearEquals)
+		for(var/obj/effect/blob/L in orange(nearEquals, T))
+			if(L.type == blobType)
+				src << "<span class='warning'>There is a similar blob nearby, move more than [nearEquals] tiles away from it!</span>"
+				return
+	if(!can_buy(price))
 		return
-
 	B.color = blob_reagent_datum.color
-	B.change_to(/obj/effect/blob/shield)
-
-	return
+	var/obj/effect/blob/N = B.change_to(blobType)
+	N.overmind = src
+	N.adjustcolors(blob_reagent_datum.color)
+	return N
 
+/mob/camera/blob/verb/create_shield_power()
+	set category = "Blob"
+	set name = "Create Shield Blob (10)"
+	set desc = "Create a shield blob."
+	create_shield()
 
+/mob/camera/blob/proc/create_shield(turf/T)
+	createSpecial(10, /obj/effect/blob/shield, 0, T)
```

Comment:
You need to either add a turf argument to create_shield() on line 57 or remove the turf/T argument in the proc definition and let the createSpecial get the turf (create_shield() is never called with a turf argument so it's not needed)